scripts/components/nav.component.js

```diff
@@ -20,12 +22,35 @@ export default class {
     this.qualDropdown = new Dropdown('qual', this.callbacks.onQualSelected);
     this.viewDropdown = new Dropdown('view', this.callbacks.onViewSelected);
 
+
     new BookmarkMenu();
     new Dropdown('bookmark', this.callbacks.onQuantSelected);
     new ShareMenu();
     new Dropdown('share');
   }
 
+  _setVars() {
+    this.el = document.querySelector('.c-nav');
+
+    this.AppNav = this.el.querySelector('.app-nav');
+    this.FlowsNav = this.el.querySelector('.flows-nav');
+    this.toggleBtn = this.el.querySelector('.js-toggle-menu');
+  }
+
+  _setEventListeners() {
+    this.toggleBtn.addEventListener('click', () => this.onToggleMenu());
+  }
+
+  onToggleMenu() {
+    this.toggleBtn.classList.toggle('open');
+    this.callbacks.onToggleAppMenu();
```

Comment:
Correct me if I'm wrong but I don't think the fact that the app menu is toggled is relevant to the rest of the app, nor is it necessary to have it as a sharable URL state.
Therefore you can keep the toggling logic entirely in `nav.component`, instead of going through the whole container/action/reducer/app store/container cycle.
This is important because I'm worried about our reducers and actions getting too big, so we should only use them when necessary.